docs/guide/building-an-advanced-scene.md

```diff
@@ -4,238 +4,456 @@ type: guide
+sound. The syntax is the same as just before, but now we are using the sound
+component's properties. We set `on` to `cursor-click` so the sound is played on
+click. And we set `src` to `audio/click.ogg`, our path to a sound file:
+
+```html
+<a-plane class="link" height="1" width="1"
+         material="shader: flat; src: #cubes-thumb"
+         sound="on: cursor-click; src: audio/click.ogg"></a-plane>
+```
+
+Now we have a textured plane that plays a click sound when clicked.
+
+## Using Third-Party Components
+
+We can grab third-party components from the ecosystem, drop them into our
+scene, and use them from HTML. Components can do anything. By using components
+that other people have developed, we gain tons of power without needing to
+write our own code.
+
+We'll go through using three such third-party components: template, layout, and
+event set. First we have to include them. [k-frame][k-frame] is a component
+pack by an A-Frame core developer that conveniently includes all three of these
+components in one bundle.
+
+To drop in `k-frame`, we can download from its [dist folder][kdist] and include it
+in the `<head>` *after* A-Frame:
+
+```html
+<html>
+  <head>
+    <title>360 Image Browser</title>
+    <script src="lib/aframe.min.js"></script>
+    <script src="lib/k-frame.min.js"></script>
+  </head>
+  <body>
+    <!-- <a-scene>... -->
+  </body>
+</html>
+```
+
+### Template Component
+
+Currently, we have one link. We want to create three of them, one for each of
+our 360&deg; images.
+
+The [template component][template] integrates templating engines into A-Frame.
+This lets us do things such as encapsulate groups of entities, passing data to
+generate entities, or iteration. Since we want to turn one link into three,
+without copy-and-pasting HTML, we can use the template component.
+
+If we read the component's documentation, we see one way to define a template
+is via a script tag in `<a-assets>`. Let's make our link a template and give it
+a name via `id`:
+
+```html
+<a-assets>
+  <!-- ... -->
+  <script id="link">
+    <a-plane class="link" height="1" width="1"
+             material="shader: flat; src: #cubes-thumb"
+             sound="on: cursor-click; src: audio/click.ogg"></a-plane>
+  </script>
+</a-assets>
+```
+
+Then we can use the template to create multiple planes without much work:
+
+```html
+<a-entity template="src: #plane"></a-entity>
+<a-entity template="src: #plane"></a-entity>
+<a-entity template="src: #plane"></a-entity>
+```
+
+But then they'll all be displaying the same image texture and look the same.
+Here is where we'll need a template engine with variable
+substitution/interpolation.
+
+Let's tell the template component to use the popular [Nunjucks][nunjucks]
+engine by specifying `<script type="text/nunjucks">`. The component will
+lazy-load the template engine for us. And with Nunjucks, we define a `{{ thumb
+}}` variable in the template, which we can pass using the data attributes:
```

Comment:
link to https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Using_data_attributes